### ISSUE TITLE
Lightning payment can go into refund state when it actually succeeded

### ISSUE THREAD

**Issue by elsirion** [fedimint/fedimint#5276]
This causes the refund to fail unexpectedly and leads to bad UX. Likely fixed by #4024? cc @m1sterc001guy

**Comment by m1sterc001guy** [comment]
Yes, apologies, this keeps getting bumped. https://github.com/fedimint/fedimint/pull/4024 should fix it, I will try to get that in this week.
